lib/tests/test_revset.rs

```diff
@@ -2305,6 +2305,91 @@ fn test_evaluate_expression_remote_bookmarks() {
     );
 }
 
+#[test]
+fn test_evaluate_expression_tags() {
```

Comment:
Not for this PR: @yuja, to avoid having to duplicate so much test code here, what do you think about moving most of them to `str_util.rs` and then trusting that code and just testing the integration here?

Reply:
Sounds good. We don't need to test every detail of string patterns in test_revset.rs.